rust/worker/src/segment/record_segment.rs

```diff
@@ -599,6 +599,22 @@ impl RecordSegmentReader<'_> {
         self.id_to_data.get("", offset_id).await
     }
 
+    pub(crate) async fn get_data_and_offset_id_for_user_id(
```

Comment:
any reason to make this a method as opposed to just letting users call it?

Reply:
I felt that it will be a common workflow to get data given a user_id hence but can change, wdyt?